patitasuy/src/app/components/post-card/post-card.component.ts

```diff
@@ -1,16 +1,29 @@
 import { Component, Input, OnInit } from '@angular/core';
 import { Post } from 'src/app/interfaces/post.interface';
+import { PostsService } from 'src/app/services/posts.service';
+import { UserService } from 'src/app/services/user.service';
 
 @Component({
   selector: 'app-post-card',
   templateUrl: './post-card.component.html',
   styleUrls: ['./post-card.component.scss'],
 })
-export class PostCardComponent  implements OnInit {
+export class PostCardComponent implements OnInit {
 
   @Input() post!: Post;
 
-  constructor() { }
+  constructor(public userService: UserService) { }
 
-  ngOnInit() {}
+  getCurrentUserId() {
```

Comment:
This method will always return 0, it needs to be async, or be executed once in the constructor